backend/src/main/java/com/google/minicrm/filters/ClientAppFilter.java

```diff
@@ -68,8 +66,27 @@ public void destroy() {
    * @param url the String representation of the url
    * @return    true if the url is a valid api url, false otherwise.
    */
-  private boolean isValidUrl(String url) {
+  private boolean isValidApiUrl(String url) {
     // valid urls start with /api (for API endpoints) or /_ah (for other GCP URLs)
     return url.startsWith("/api") || url.startsWith("/_ah");
   }
+
+  private void redirectToClient(ServletRequest servletRequest, ServletResponse servletResponse,
+                                String path) throws IOException, ServletException {
+    RequestDispatcher dispatcher;
+    dispatcher = servletRequest.getRequestDispatcher(localizePath(path));
+    dispatcher.forward(servletRequest, servletResponse);
+  }
+
+  private String localizePath(String url) {
+    if (url.startsWith("/hi")) {
```

Comment:
maybe something like if (url.startsWith("/hi") || url.startsWith("/es") ....) return url.substring(0, 3) + "/index.html"?
also, no need for the last "else", can just return "/en/index.html" at the bottom, since no other option is possible.